### PR TITLE
Docs: Update readme for legend describing rules icons (refs #4355)

### PR DESCRIPTION
**Pull request by kaicataldo**
As discussed in https://github.com/eslint/eslint.github.io/pull/180, here is the updated readme doc. Let me know what I need to fix, if anything.

### CODE REVIEW

**Comment by kaicataldo** [comment]
I'm running into an issue where the CI checks are feeling because of trailing whitespace, but this seems to be necessary to put a new line between the legend lines. Thoughts?

**Comment by ilyavolodin** [comment]
Linter is complaining about trailing whitespace, not linebreak. You have an extra space after

`recommended` = (recommended)

**Comment by kaicataldo** [comment]
I don't think I explained it very well so I stripped the whitespace to make it clearer. With just a line break, those two lines get rendered on the same line: https://github.com/kaicataldo/eslint/blob/fixes4355/docs/rules/README.md

**Comment by kaicataldo** [comment]
For context: "When you do want to insert a <br /> break tag using Markdown, you end a line with two or more spaces, then type return." [from the Markdown docs](http://daringfireball.net/projects/markdown/syntax#p).

**Comment by ilyavolodin** [comment]
Ahh.. Now I understand. I think that's fine to render on the same line. Thanks.

File: docs/rules/README.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+Rules in ESLint are divided into several categories to help you better understand their value. All rules are disabled by default. ESLint recommends some rules to catch common problems, and you can use these recommended rules by including `extends: "eslint:recommended"` in your configuration file. The rules that will be enabled when you inherit from `eslint:recommended` are indicated by a checkmark icon below. For more information on how to configure rules and use `extends`, please see the [configuration documentation](../user-guide/configuring.md).
 
-Some rules are fixable using the `--fix` command line flag. Those rules are marked as "(fixable)" below.
+Some rules are fixable using the `--fix` command line flag. Those rules are marked with a wrench icon below.
+
+`recommended` = (recommended)
+`fixable` = (fixable)
```
Comment (nzakas) [inline docs/rules/README.md]: Why are these in code text?

**Comment by nzakas** [comment]
I'm not a fan of this change. IMHO, it makes it harder to figure out what's going on. Describing the icons is a bad idea (" wrench icon") because of language issues some users might have.  I think the original text is clearer because it shows the icons in context with the description. I'd like to revert this, we can always add a legend underneath the original text.

File: docs/rules/README.md (hunk 1 above)
Comment (kaicataldo) [inline docs/rules/README.md]: I added that for the styling, but you're right, it doesn't make much sense semantically

**Comment by kaicataldo** [comment]
@nzakas I honestly agree with you - I was asked to add the legend while making the original PR. No qualms reverting this on my end!

**Comment by nzakas** [comment]
I think the legend is a good idea, I just wouldn't change the original text. Rather, just out a legend right under it.

**Comment by kaicataldo** [comment]
@nazakas Alright, I'll open a PR.

**Pull request merged**